parquet/src/main/scala/magnolify/parquet/ParquetField.scala

```diff
@@ -339,15 +343,16 @@ object ParquetField {
           v.foreach(t.writeGroup(c, _)(cm))
         }
 
-      override def newConverter: TypeConverter[C[T]] = {
-        val buffered = t.newConverter
+      override def newConverter(writerSchema: Type): TypeConverter[C[T]] = {
+        val buffered = t
+          .newConverter(writerSchema)
           .asInstanceOf[TypeConverter.Buffered[T]]
           .withRepetition(Repetition.REPEATED)
         val arrayConverter = new TypeConverter.Delegate[T, C[T]](buffered) {
           override def get: C[T] = inner.get(fc.fromSpecific)
         }
 
-        if (hasAvroArray) {
+        if (Schema.hasGroupedArray(writerSchema)) {
```

Comment:
I guess this would throw an error if the schema mixed grouped and non-grouped array types since it checks for the presence of a grouped array in the entire schema, rather than for the specific field.... but we don't have access to `CaseMapper` here so we couldn't easily isolate the specific field schema. The original approach with the `AvroCompat` import was also all-or-nothing so this shouldn't functionally be a change in behavior